### PR TITLE
Parse GSheets data

### PR DESCRIPTION
**Pull request by takecare**
#### Problem

As we're now able to get data from GSheets, we need to be able to parse it appropriately. A key aspect of the parsing process is cleaning up some extra data GSheets adds - the column name (only on the values of the 2nd column).

Example:

| Name | Occupation |
| --- | --- |
| Mark | Loan manager |
| Super Hans | Musician |
| Jez | Unemployed |

Each of the cells for the "Occupation" column will come back as `Occupation: Loan manager`, `Occupation: Musician` and `Occupation: Unemployed`.

Given the complex json model we're getting a sheet document as, the data we need to get is buried deep into nested objects. 

Example:

```
{
    (...)
    "feed": {
        (...)
        "entry": [
            {
                (...)
                "title": {
                    "type": "text",
                    "$t": "Mark"
                },
                "content": {
                    "type": "text",
                    "$t": "Occupation: Loan manager"
                },
            },
            {
                (...)
                "title": {
                    "type": "text",
                    "$t": "Super Hans"
                },
                "content": {
                    "type": "text",
                    "$t": "Occupation: Musician"
                },
            },
            {
                (...)
                "title": {
                    "type": "text",
                    "$t": "Jez"
                },
                "content": {
                    "type": "text",
                    "$t": "Occupation: Unemployed"
                },
            }
        ]
    }
}
```

This led to the need of having somewhat redundant objects such as the `Content` one, which would wrap both `"title"` and `"content"`, above. Ideally we should be able to skip that and just store the data in an `Entry` instance.

#### Solution

As we're already relying on `Gson` to parse the json data for us, a custom `EntryDeserializer` was added so we can drop some unnecessary model classes - `Content` - and parse their data directly into `Entry`s.

A class responsible for removing the extra data (mentioned above) was also added.

##### Test(s) added

Yes.

### CODE REVIEW

File: sheets/src/main/java/com/novoda/github/reports/sheets/sheet/Entry.java [hunk 1]
```diff
@@ -1,30 +1,25 @@
...
-
 public class Entry {
 
-    @SerializedName("title")
-    private Content key;
-
-    @SerializedName("content")
-    private Content value;
+    private String title;
+    private String content;
```
Comment (xrigau) [inline sheets/src/main/java/com/novoda/github/reports/sheets/sheet/Entry.java]: these 2 should be `final` now

File: sheets/src/test/java/com/novoda/github/reports/sheets/network/SheetsServiceClientTest.java [hunk 2]
```diff
@@ -38,13 +43,15 @@
     public void setUp() throws Exception {
         initMocks(this);
 
+        when(mockValueRemover.removeFrom(any(Entry.class))).thenAnswer(invocation -> invocation.getArgument(0));
```
Comment (xrigau) [inline sheets/src/test/java/com/novoda/github/reports/sheets/network/SheetsServiceClientTest.java]: you might as well create a test remover that doesn't remove at this point 😉

Comment (takecare) [inline sheets/src/test/java/com/novoda/github/reports/sheets/network/SheetsServiceClientTest.java]: same effort really 😄 but i'll go for it 'cause it'll read easier!

Comment (takecare) [inline sheets/src/test/java/com/novoda/github/reports/sheets/network/SheetsServiceClientTest.java]: actually, i'll have to drop this in the next pr 😞

**Comment by xrigau** [comment]
time to 🚢

**Pull request merged**